nodes/src/nodes/branch/branch_engine.py

```
@@ -0,0 +1,389 @@
+        text_lower = text.lower()
+        keyword_list = [k.strip().lower() for k in keywords.split(',') if k.strip()]
+
+        if not keyword_list:
+            return {'matched': False, 'condition': 'contains', 'details': 'no valid keywords'}
+
+        found = [kw for kw in keyword_list if kw in text_lower]
+
+        if mode == 'all':
+            matched = len(found) == len(keyword_list)
+        else:
+            matched = len(found) > 0
```

Ruff (0.15.7)

[warning] Use ternary operator `matched = len(found) == len(keyword_list) if mode == 'all' else len(found) > 0` instead of `if`-`else`-block

Replace `if`-`else`-block with `matched = len(found) == len(keyword_list) if mode == 'all' else len(found) > 0`

(SIM108)